### DISCUSSION TITLE
Question about FileDialog

### DISCUSSION THREAD

**Discussion by JLG-ISC** [LesFerch/FileDialog#3] [Q&A] [answered]
How does the app work? Does it leverage explorer.exe. The reason I ask, I recently deployed 2019 RDS Farm, which runs PowerPoint 2019 as a published app. The problem is that opening a pps file from within PowerPoint does not open it in show mode.  I am being asked to publish explorer.exe as a separate published app.  I am hesitant to do this.  I am hoping the FileDialog can be called as a published app giving users the ability to double click a pps file and have it open in show mode.  But I want to be sure there is benefits in running FileDialog over explorer.exe with regard to RDSH stability and resource utilization.

Please advise

**Comment by LesFerch** [comment, marked as answer]
FileDialog does not use explorer.exe. Any app that has a standard file open/save dialog is using a built-in OS feature that provides a a dialog that looks like a mini version of the Windows 10 Explorer (note that file dialogs look the same on Windows 11). File open/save dialogs may use a DLL that is also used by Explorer, but if you `taskkill /f /im explorer.exe` FileDialog will not be affected.

Since Explorer also provides the desktop and taskbar, it's not a simple standalone app like other file managers. I don't know if there would be an issue setting up Explorer as a published app, but if there is, I suppose using FileDialog.exe (or any other standalone file manager) would be good solution.

Note that if you kill explorer with `taskkill /f /im explorer.exe` and then run explorer with a path provided as an argument, such as `explorer %documents%`, explorer launches as a file manager only.
